Fixes #18902 - Enable autocomplete for container image search

bastilian opened the pull request:
This enables the autocomplete for the image name search and
improves the usability of searching for container images on
Docker Hub or an external Registry.

It also improves the form a bit and aligns everything nicer:

![screen shot 2017-03-27 at 20 56 23](https://cloud.githubusercontent.com/assets/7757/24373020/41484cc6-1330-11e7-8ffc-717e9d6880ff.png)
![screen shot 2017-03-27 at 20 56 39](https://cloud.githubusercontent.com/assets/7757/24373022/41507dec-1330-11e7-82c6-82f212ad4d3b.png)
![screen shot 2017-03-27 at 20 56 55](https://cloud.githubusercontent.com/assets/7757/24373021/414d74f8-1330-11e7-93dd-520e6a7d8fda.png)

This is dependent on #186 being merged to master first.

bastilian commented:
@Rohoover this is now the "polished" version of the autocompletion and availability check for container images. Everything is now better aligned. Is there anything else you'd like to see or have differently?

ohadlevy commented on app/assets/javascripts/foreman_docker/container_image_search.js:
```diff
@@ -0,0 +1,203 @@
+      callback(data);
+    }.bind(this))
+    .error(function(result) {
+      $.jnotify(result.responseText, 'error', true);
```
please use the notifify function instead of calling jnotify directly, as it will be deprecated.

bastilian commented:
@ohadlevy replaced `$.jnotify()` now with `notify()`

Rohoover commented:
Only two comments:

1) Remove icons in tabs. They are identical so have reduced value and I'm not sure the icons meaning aligns with the content.

2) Can we align the back and next buttons according to PatternFlys Form Within a Page standard?
http://www.patternfly.org/pattern-library/forms-and-controls/forms/

bastilian commented:
@Rohoover the icons on the tabs are now removed and the buttons are aligned as Patternfly suggests:

![create container](https://cloud.githubusercontent.com/assets/7757/24415959/6c22f78a-13e3-11e7-9994-f964f13cb097.png)

I changed the wording of the buttons to "Next Step" and "Previous Step", to not end up with this:

![screen shot 2017-03-28 at 17 50 13](https://cloud.githubusercontent.com/assets/7757/24415960/6c2499c8-13e3-11e7-9263-2ffd1a0fa5b7.png)

The buttons are used on each step like this.

Rohoover commented:
![bitmoji](https://render.bitstrips.com/v2/cpanel/7258102-131321122_2-s4-v1.png?transparent=1&palette=1&width=246)

bastilian commented:
@dLobatog this is now rebased on the current master and includes now only changes for the autocomplete feature. If you want, can you give it another look and see if the issues you've briefly mentioned in IRC still occur?

dLobatog commented:
@bastilian I think I was having some kind of DNS problem as I can't reproduce the timeouts now. Anyway, it works great, thanks 👍 !

@Rohoover The only problem I see is that "Next" and "Previous" buttons are now reversed. "Next" moves you to the next step on the wizard which is conceptually located *right* of the current step. However "Next" is now on the left, and "Previous" is now on the right, which is confusing IMO (I myself clicked it wrong a few times). Even https://www.patternfly.org/pattern-library/communication/wizard/ says the buttons should be laid out the opposite way they are now.

Rohoover commented:
@dLobatog 

Ah yes, you are correct. I miss took this for a form and not a wizard. Please follow the PF recommendation. 

The PF wizard doesn't address this on the site but I believe the button should be aligned left (as they currently are) as opposed to right since it is not a modal. 

Thanks for catching this!

bastilian commented:
@Rohoover & @dLobatog I've switched the wizard buttons now:

![screen shot 2017-04-24 at 08 36 22](https://cloud.githubusercontent.com/assets/7757/25325022/76cf689e-28c9-11e7-89cb-37c6eae8d8ac.png)

Is this the way it should be or should they be moved to the right side of the form?

dLobatog commented:
@bastilian Yes, thanks!

bastilian commented:
[test]

xprazak2 commented:
@bastilian, could you rebase pls?

bastilian commented:
@xprazak2 rebased. the test will be blocked though, since the fix in master is not merged yet. see https://github.com/theforeman/foreman/pull/4500

bastilian commented:
[test]

bastilian commented:
test failures on sqlite seem to be an intermittent issue.

bastilian commented:
@dLobatog rebased. I'll merge when CI is green.

dLobatog commented:
Retriggering test_plugin_foreman_docker_master in order to make it green

bastilian commented:
@dLobatog eureka! it's green!
